app/gui/view/component-browser/component-group-view/src/lib.rs

```diff
@@ -0,0 +1,224 @@
+//! An EnsoGL implementation of a Component Group View.
+//!
+//! A Component Group View is displayed as a header (containing the name of a group of Enso
+//! components) and a list of entries below it (names of the components belonging to that group).
+
+#![recursion_limit = "512"]
+// === Standard Linter Configuration ===
+#![deny(non_ascii_idents)]
+#![warn(unsafe_code)]
+// === Non-Standard Linter Configuration ===
+#![warn(missing_copy_implementations)]
+#![warn(missing_debug_implementations)]
+#![warn(missing_docs)]
+#![warn(trivial_casts)]
+#![warn(trivial_numeric_casts)]
+#![warn(unused_import_braces)]
+#![warn(unused_qualifications)]
+
+use ensogl_core::prelude::*;
+
+use enso_frp as frp;
+use ensogl_core::application::Application;
+use ensogl_core::data::color::Rgba;
+use ensogl_core::display;
+use ensogl_core::display::shape::*;
+use ensogl_gui_component::component;
+use ensogl_gui_component::component::ComponentView;
+use ensogl_hardcoded_theme::application::component_browser::component_group as theme;
+use ensogl_list_view as list_view;
+use ensogl_list_view::entry;
+use ensogl_list_view::ListView;
+use ensogl_text as text;
+
+
+
+// ==========================
+// === Shapes Definitions ===
+// ==========================
+
+
+// === Header Background ===
+
+/// The background of the Component Group View's header.
+pub mod header_background {
+    use super::*;
+
+    ensogl_core::define_shape_system! {
+        above = [list_view::selection];
+        (style:Style, color:Vector4) {
+            let sprite_width: Var<Pixels> = "input_size.x".into();
+            let sprite_height: Var<Pixels> = "input_size.y".into();
+            let color = Var::<Rgba>::from(color);
+            let shape = Rect((&sprite_width, &sprite_height)).fill(color);
+            shape.into()
+        }
+    }
+}
+
+
+
+// ===========
+// === FRP ===
+// ===========
+
+ensogl_core::define_endpoints_2! {
+    Input {
+        set_header_text(String),
+        set_entries(entry::AnyModelProvider<entry::Label>),
+        set_background_color(Rgba),
+        set_size(Vector2),
+    }
+    Output {}
+}
+
+impl component::Frp<Model> for Frp {
+    fn init(api: &Self::Private, _app: &Application, model: &Model, style: &StyleWatchFrp) {
+        let network = &api.network;
+        let input = &api.input;
+        let header_text_size = style.get_number(theme::header::text::size);
+        frp::extend! { network
+
+            // === Geometry ===
+
+            let header_geometry = HeaderGeometry::from_style(style, network);
```

Comment:
Please, try making the file readable from top-to-bottom. `HeaderGeometry` was not introduced yet. Move it above these definitions, if possible, please.